FreeRTOS_Sockets.c

```diff
@@ -1439,7 +1439,7 @@ BaseType_t FreeRTOS_closesocket( Socket_t xSocket )
 {
     BaseType_t xResult;
 
-    #if ( ipconfigUSE_TCP == 1 ) && ( ipconfigUSE_CALLBACKS == 1 )
+    #if ( ipconfigUSE_TCP == 1 ) || ( ipconfigUSE_CALLBACKS == 1 )
         FreeRTOS_Socket_t * pxSocket = ( FreeRTOS_Socket_t * ) xSocket;
```

Comment:
Thank you for this change. But looking a second time, I think that testing `ipconfigUSE_TCP` is not needed:
~~~c
    #if ( ipconfigUSE_CALLBACKS == 1 )
        FreeRTOS_Socket_t * pxSocket = ( FreeRTOS_Socket_t * ) xSocket;
    #endif
~~~
because when `ipconfigUSE_CALLBACKS` are disabled, `pxSocket ` will not be referred too?

Reply:
You are correct that makes more sense, fixed. Also, I believe the UDP receive callback is never called?